tests/CMakeInstallTests/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# This file is used to test the CMake installation and library finding logic
+# for the package
+cmake_minimum_required(VERSION 3.16)
+project( OSQP_test_install )
+
+# The test install is up two levels
+set( OSQP_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/../../../install" )
+
+find_package( OSQP )
```

Comment:
I have added tests to ensure that the version check now works by passing in both version 0.99 and version 1.0. The 0.99 test should work (since that is the version it is built with) and the 1.0 check should fail because the major version doesn't match in the strategy we use.